protocol/message.go

```diff
@@ -52,48 +52,77 @@ type MonitoringRequest struct {
 }
 
 // Response messages
-type Response interface{}
-
-type ErrorResponse struct {
-	Error ErrorCode
+type Response struct {
```

Comment:
In that case, we should drop `Type` from the response altogether.  `ExpectingDirProofResponse` is hinting at the fact that we already know what type to expect for responses anyways.